packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -364,16 +347,46 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
                     : `[]`,
                   { contentOnly: true }
                 )
+                rewroteMarkerStartPos.add(markerStartPos)
               }
             }
-            updateChunkCode(s)
           }
 
           // there may still be markers due to inlined dynamic imports, remove
           // all the markers regardless
-          const s = new MagicString(chunk.code)
-          s.replace(preloadMarkerRE, 'void 0')
-          updateChunkCode(s)
+          let markerStartPos = code.indexOf(preloadMarkerWithQuote)
+          do {
+            if (!rewroteMarkerStartPos.has(markerStartPos)) {
+              s.overwrite(
+                markerStartPos,
+                markerStartPos + preloadMarkerWithQuote.length,
+                'void 0',
+                { contentOnly: true }
+              )
+            }
+
+            markerStartPos = code.indexOf(
+              preloadMarkerWithQuote,
+              markerStartPos + preloadMarkerWithQuote.length
+            )
+          } while (markerStartPos >= 0)
```

Comment:
Should we check for the `markerStartPos >= 0` in case there aren't any markers? And if that isn't the case, I still find it easier to read. 
```suggestion
          let markerStartPos = code.indexOf(preloadMarkerWithQuote)
          while (markerStartPos >= 0) {
            if (!rewroteMarkerStartPos.has(markerStartPos)) {
              s.overwrite(
                markerStartPos,
                markerStartPos + preloadMarkerWithQuote.length,
                'void 0',
                { contentOnly: true }
              )
            }

            markerStartPos = code.indexOf(
              preloadMarkerWithQuote,
              markerStartPos + preloadMarkerWithQuote.length
            )
          }
```

Reply:
Good catch!